prow/job/sippy.py

```diff
@@ -25,42 +26,72 @@ def __init__(self, host, port=443, secure=True):
 
         self._adapter = HTTPAdapter(max_retries=retry_strategy)
 
-    def _request(self, url, params):
+    def _get_session(self):
         session = requests.Session()
-        session.mount("http://", self._adapter)
-        session.mount("https://", self._adapter)
+        session.mount(
+            f"{'https' if self._secure else 'http'}://", self._adapter)
+        return session
 
-        response = session.get(url, params=params, verify=False)
+    def _get_request(self, url, params, data=None):
+        session = self._get_session()
+        response = session.get(url, params=params, verify=False, data=data)
+        response.raise_for_status()
+
+        return response.json()
+
+    def _post_request(self, url, payload, headers=None):
+        if not payload:
+            raise ValueError("request payload should not be empty")
+
+        session = self._get_session()
+        response = session.post(url, data=payload, headers=headers)
         response.raise_for_status()
 
         return response.json()
 
     def health_check(self, release):
         url = f"{self._base_url}/health"
-        return self._request(url, ParamBuilder().release(release).done())
+        return self._get_request(url, ParamBuilder().release(release).done())
 
     def query_jobs(self, params):
         url = f"{self._base_url}/jobs"
-        return self._request(url, params)
+        return self._get_request(url, params)
 
     def query_tests(self, params):
         url = f"{self._base_url}/tests"
-        return self._request(url, params)
+        return self._get_request(url, params)
 
     def query_component_readiness(self, params):
         url = f"{self._base_url}/component_readiness"
-        return self._request(url, params)
+        return self._get_request(url, params)
 
     def query_variant_status(self, params):
         url = f"{self._base_url}/variants"
-        return self._request(url, params)
+        return self._get_request(url, params)
+
+    def query_risk_analysis(self, payload=None, job_run_id=None):
+        if payload is None and job_run_id is None:
+            raise ValueError("payload and job_run_id are all empty")
+
+        url = f"{self._base_url}/jobs/runs/risk_analysis"
+        if job_run_id:
+            # send get request
+            return self._get_request(url, ParamBuilder().prow_job_run_id(job_run_id).done())
+
+        if payload:
```

Comment:
I prefer to use the `job_data` since the `payload` is confusing for the user.

Reply:
done